zenoh-jni/src/subscriber.rs

```diff
@@ -130,13 +136,23 @@ pub(crate) unsafe fn declare_subscriber(
                 }
             };
 
-            let key_expr_ptr = Arc::into_raw(Arc::new(sample.key_expr));
+            let key_expr_str = match env.new_string(sample.key_expr.to_string()) {
+                Ok(key_expr_str) => key_expr_str,
+                Err(err) => {
+                    tracing::error!(
+                        "Could not create a JString through JNI for the Sample key expression. {}",
+                        err
+                    );
+                    return;
+                }
+            };
+
```

Comment:
nit: There may be a nicer way to write all these matches that have returns, but id need to think about it